core/src/types.ts

```diff
@@ -108,23 +114,35 @@ declare interface MarketDataBase extends Partial<ExchangeFees> {
     errorMessage?: any;
 }
 
-declare interface MarketDataRegular extends MarketDataBase, Omit<RegularCalleeConfig, 'callee'> {}
+export declare interface Pool {
+    addresses: string[];
+    fee: number;
+}
+export declare interface MarketDataRegular extends MarketDataBase, Omit<RegularCalleeConfig, 'callee'> {
+    pools: Pool[]
+}
+
+export declare interface MarketDataUniswapV2LpToken extends MarketDataBase, Omit<UniswapV2LpTokenCalleeConfig, 'callee'> {}
 
-declare interface MarketDataUniswapV2LpToken extends MarketDataBase, Omit<UniswapV2LpTokenCalleeConfig, 'callee'> {}
+export declare interface MarketDataUniswapV3Automatic extends MarketDataBase, Omit<UniswapV3AutoRouterCalleeConfig, 'callee'> {
+    route: string[];
+    pools: Pool[];
+}
```

Comment:
Pools should now be part of every market data, otherwise you risk to spread the logic of determining between automatic and hardcoded everywhere, which is a lot of edits and a lot of ifs